src/VecSim/algorithms/hnsw/hnswlib.h

```diff
@@ -60,14 +63,14 @@ class HierarchicalNSW : public VecsimBaseObject {
 
     // Index state
     size_t cur_element_count;
-    int max_id;
-    int maxlevel_;
+    tableint max_id;
+    size_t maxlevel_;
```

Comment:
Why `maxlevel_` is `size_t` and not `unsigned int`? It shouldn't be larger than the number of ids, and we use `unsigned int` for that...